pkg/daemon/ceph/osd/volume.go

```diff
@@ -79,9 +80,25 @@ func (a *OsdAgent) initializeDevices(context *clusterd.Context, devices *DeviceO
 		strconv.Itoa(a.storeConfig.OSDsPerDevice),
 	}...)
 
-	// ceph-volume is soon implementing a parameter to specify the "fast devices", which correspond to the "metadataDevice" from the
-	// crd spec. After that is implemented, we can implement this. In the meantime, we fall back to use rook's partitioning.
+	if a.storeConfig.StoreType == config.Bluestore && a.storeConfig.DatabaseSizeMB > 0 {
+		if a.storeConfig.DatabaseSizeMB < 1024 {
+			// ceph-volume will convert this value to ?G. It needs to be > 1G to invoke lvcreate.
+			logger.Infof("skipping databaseSizeMB setting. For it should be larger than 1024MB.")
+		} else {
+			batchArgs = append(batchArgs, []string{
+				databaseSizeFlag,
+				strconv.Itoa(a.storeConfig.DatabaseSizeMB * 1024 * 1024), // ceph-volume takes in this value in bytes
```

Comment:
I believe this will overflow the int when you hit around 2GB, so we likely need an int64 conversion

Reply:
Good catch, will fix. 💯